scylla/src/transport/cluster.rs

```diff
@@ -376,7 +376,17 @@ impl ClusterWorker {
             // Perform the refresh
             debug!("Requesting topology refresh");
             last_refresh_time = Instant::now();
-            let refresh_res = self.perform_refresh().await;
+            let refresh_res =
+                match tokio::time::timeout(refresh_duration, self.perform_refresh()).await {
```

Comment:
AFAIK metadata refresh is performed at the session startup. If there are no nodes available from the peers list, it looks like the new loop in `read_metadata` will spin (preemptibly) trying to connect and will only give up after `refresh_duration`, which is 60s by default. Could you check if that is the case?

Reply:
Yeah, it is. Here's an idea - could you manually merge only the second patch that adds more debug prints for now? And I'll work on reworking the patch that changes the refresh logic in a later time

Reply:
(I checked, it applies cleanly on origin/main)

Reply:
OK, done.